lib/pages/api_key_dialog.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+
+import '../l10n/l10n.dart';
+import '../main.dart';
+import '../widgets/options_list_item.dart';
+
+class ApiKeyDialog extends StatelessWidget {
+  final apiKeyIndicatorColor =
+      authStore.apiKey == null ? Colors.red : Colors.green;
+  final apiKeyIndicatorText = authStore.apiKey == null
+      ? L10nStrings.apiKeyDialog_apiKeyIsNotSaved
+      : L10nStrings.apiKeyDialog_apiKeyIsSaved;
+  final TextEditingController apiKeyTextFieldEditingController =
```

Comment:
You cant use TextEditingController in a stateless widget, it will get recreated on each render. Also you need to dispose the controller when the widget is unmounted